File: CSE3063F24P1_GRP2-main/src/Password.java
public class Password {
    // Attribute
    private String password;

    // Constructor
    public Password(String password) {
        this.password = password;
    }

    // Checks if the password is secure
    public boolean isSecurePassword() {
        // Check if the password is null or empty
        if (password == null || password.isEmpty()) {
            return false;
        }

        // Check password length (minimum 8 characters)
        if (password.length() < 8) {
            return false;
        }

        // Check for at least one digit
        if (!password.matches(".*\\d.*")) {
            return false;
        }

        // Check for at least one special character
        if (!password.matches(".*[@#$%^&+=!].*")) {
            return false;
        }

        return true; // If all conditions are met (no letter condition required)
    }

    // Getter and Setter
    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }
}

File: Iteration3/Python/Password.py
class Password:


    def __init__(self, password):
        self.password = password

    def is_secure_password(self):

        if len(self.password) < 8:
            return False
        if not any(char.islower() for char in self.password):
            return False
        if not any(char.isupper() for char in self.password):
            return False
        if not any(char.isdigit() for char in self.password):
            return False
        if not any(char in "!@#$%^&*()-_=+[{]}\\|;:'\",<.>/?`~" for char in self.password):
            return False
        return True

    def get_password(self):

        return self.password

    def set_password(self, password):

        self.password = password
